Mathlib/NumberTheory/LSeries/Deriv.lean

```diff
@@ -0,0 +1,249 @@
+/-
+Copyright (c) 2024 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+import Mathlib.Analysis.Calculus.SmoothSeries
+import Mathlib.Analysis.Complex.CauchyIntegral
+import Mathlib.Analysis.Convex.Complex
+import Mathlib.Data.Complex.ExponentialBounds
+import Mathlib.NumberTheory.LSeries.Convergence
+import Mathlib.Analysis.SpecialFunctions.Pow.Deriv
+
+/-!
+# Differentiability and derivatives of L-series
+
+## Main results
+
+* We show that the `LSeries` of `f` is differentiable at `s` when `re s` is greater than
+  the abscissa of absolute convergence of `f` (`LSeries.hasDerivAt`) and that its derivative
+  there is the negative of the `LSeries` of the point-wise product `log * f` (`LSeries.deriv`).
+
+* We prove similar results for iterated derivatives (`LSeries.iteratedDeriv`).
+
+* We use this to show that `LSeries f` is holomorphic on the right half-plane of
+  absolute convergence (`LSeries.analyticOn`).
+
+## Implementation notes
+
+We introduce `LSeries.logMul` as an abbreviation for the point-wise product `log * f`, to avoid
+the problem that this expression does not type-check. Similarly, we make use of the abbreviation
+`LSeries.logPowMul n f` for `log^n * f`.
+-/
+
+-- Should this go into its own file (because it needs `EReal`s, it doesn't fit well
+-- in some existing file), together with versions for the left, upper and lower half-planes?
+/-- An open right half-plane is open in the complex plane. -/
+lemma Complex.isOpen_rightHalfPlane (x : EReal) : IsOpen {z : ℂ | x < z.re} :=
+  isOpen_lt continuous_const <| EReal.continuous_coe_iff.mpr continuous_re
```

Comment:
Here's a suggestion. One could add the lemma
````
lemma EReal.isOpen_gt (x : EReal) {α : Type*} [TopologicalSpace α] {f : α → ℝ} (hf : Continuous f) :
    IsOpen {a : α | x < f a} :=
  isOpen_lt continuous_const <| EReal.continuous_coe_iff.mpr hf
````
to `Mathlib.Topology.Instances.EReal` (and its obvious mirror-image). Then the uses of `isOpen_rightHalfPlane` can be replaced with things like `(EReal.isOpen_gt _ continuous_re)` (at line 199 and 229) or the dot-notation `... <| (abscissaOfAbsConv f).isOpen_gt continuous_re` (for the last line). Just a suggestion, feel free to ignore if you prefer another approach.

Reply:
I think it is nice to have this version available, so I'd like to keep it. But perhaps the name should be changed to `EReal.isOpen_complexRightHalfPlane`? This would allow for dot notation.

Reply:
IMHO, dot-notation is a nice thing to have, but we shouldn't let it override the use of namespaces to keep related results together – a lemma which is specifically about complex numbers should not go in the `EReal` namespace.

Reply:
OK. I'll leave it as is.

Reply:
Would you be happy with renaming `Complex.isOpen_rightHalfPlane` to `Complex.isOpen_re_gt_EReal` and moving it to either a new file, or one of the existing files, in the `Analysis/Complex` directory?